probe/docker/container.go

```diff
@@ -220,6 +222,48 @@ func (c *container) ports(localAddrs []net.IP) report.StringSet {
 	return report.MakeStringSet(ports...)
 }
 
+func (c *container) cpuPercentMetric() report.Metric {
+	result := report.MakeMetric()
+	if len(c.pendingStats) < 2 {
+		return result
+	}
+
+	previous := c.pendingStats[0]
+	for _, s := range c.pendingStats[1:] {
+		// Copies from docker/api/client/stats.go#L205
+		cpuDelta := float64(s.CPUStats.CPUUsage.TotalUsage - previous.CPUStats.CPUUsage.TotalUsage)
+		systemDelta := float64(s.CPUStats.SystemCPUUsage - previous.CPUStats.SystemCPUUsage)
+		cpuPercent := 0.0
+		if systemDelta > 0.0 && cpuDelta > 0.0 {
+			cpuPercent = (cpuDelta / systemDelta) * float64(len(s.CPUStats.CPUUsage.PercpuUsage)) * 100.0
+		}
+		result = result.Add(s.Read, cpuPercent)
+		previous = s
+	}
+	return result
+}
+
+func (c *container) metric(f func(*docker.Stats) float64) report.Metric {
+	result := report.MakeMetric()
+	for _, s := range c.pendingStats {
+		result = result.Add(s.Read, f(s))
+	}
+	return result
+}
+
+func (c *container) metrics() report.Metrics {
+	result := report.Metrics{
+		MemoryUsage:   c.metric(func(s *docker.Stats) float64 { return float64(s.MemoryStats.Usage) }),
```

Comment:
The idea is that would be reusable for other (absolute) docker metrics, if we find any.